scripts/packaging/createupmpackages.ps1

```diff
@@ -3,38 +3,53 @@
     Builds the Mixed Reality Toolkit Unity Package Manager (UPM) packacges.
 .DESCRIPTION
     Builds UPM packages for the Mixed Reality Toolkit.
-.PARAMETER NodejsVersion
-    The desired version of node.js to use when packaging. If not specified, a known good version will be used.
-.PARAMETER OutputDirectory
-    Where should we place the output? Defaults to ".\artifacts"
-.PARAMETER PackageVersion
-    What version of the artifacts should we build? If unspecified, the highest
-    git tag pointing to HEAD is searched. If none is found, an error is reported.
 .PARAMETER ProjectRoot
     The root folder of the project.
+.PARAMETER OutputDirectory
+    Where should we place the output? Defaults to ".\artifacts"
+.PARAMETER Version
+    What version of the artifacts should we build?
+.PARAMETER BuildNumber
+    The build number to append to the version. Note: This value is only used if the IsPreview parameter is set to true.
+.PARAMETER IsOfficial
+    Are we creating official packages? If not, the version of the artifiacts will be formatted as "<$Version>-preview.<BuildNumber>".
 #>
 param(
-    [ValidatePattern("^\d+\.\d+\.\d+")]
-    [string]$NodejsVersion = "12.18.0",
+    [string]$ProjectRoot,
     [string]$OutputDirectory = ".\artifacts\upm",
-    [ValidatePattern("^\d+\.\d+\.\d+-?[a-zA-Z0-9\.]*$")] # todo - format of d.d.d[-preview.0-9.0-9]
-    [string]$PackageVersion,
-    [string]$ProjectRoot
+    [ValidatePattern("^\d+\.\d+\.\d+-?[a-zA-Z0-9\.]*$")]
+    [string]$Version,
+    [ValidatePattern("^\d+?[\.\d+]*$")]
+    [string]$BuildNumber,
+    [bool]$IsOfficial = $False
 )
 
-if (-not $PackageVersion) {
-        throw "Unknown package version. Please specify -PackageVersion when building."
+$startPath = "$(Get-Location)"
+
+if (-not $ProjectRoot) {
+    throw "Unknown project root path. Please specify -ProjectRoot when building."
+}
+$ProjectRoot = Resolve-Path -Path $ProjectRoot
+Write-Output "Project root: $ProjectRoot"
+
+if (-not $Version) {
+    throw "Unknown package version. Please specify -Version when building."

```

Comment:
"-Version is a required flag and is missing"

Reply:
(I think the "Unknown" is still somewhat ambiguous because it could have been provided but maybe it's not a known version number).

In general for things like this where the flag isn't provided, just saying "X is required and hasn't been set, please set it dude" is more explicit

Reply:
i will update the messages

Reply:
fixed in all 3 scripts